programs/fileio.c

```diff
@@ -862,6 +862,193 @@ int FIO_compressFilename(const char* dstFileName, const char* srcFileName,
+        magicNumber = MEM_readLE32(magicNumberBuffer);
+        if(magicNumber==ZSTD_MAGICNUMBER){
+            BYTE frameHeaderDescriptor;
+            int totalFrameHeaderBytes;
+            BYTE* frameHeader;
+            int lastBlock = 0;
+            size_t readBytes = fread(&frameHeaderDescriptor, 1, 1, srcFile);
+            info->numActualFrames++;
+            if(readBytes != 1){
+                DISPLAY("There was an error with reading frame header descriptor\n");
+                exit(1);
+            }
+            /* calculate actual frame header size */
+            totalFrameHeaderBytes = calcFrameHeaderSize(frameHeaderDescriptor);
+
+            /* reset to beginning of from and read entire header */
+            fseek(srcFile, -5, SEEK_CUR);
+            frameHeader = (BYTE*)malloc(totalFrameHeaderBytes);
+            readBytes = fread(frameHeader, 1, totalFrameHeaderBytes, srcFile);
+            if(readBytes != (size_t)totalFrameHeaderBytes){
+                DISPLAY("There was an error reading the frame header\n");
+                exit(1);
+            }
+
+            /* get decompressed file size */
+            {
+                U64 additional = ZSTD_getFrameContentSize(frameHeader, totalFrameHeaderBytes);
+                if(additional!=ZSTD_CONTENTSIZE_UNKNOWN && additional!=ZSTD_CONTENTSIZE_ERROR){
+                    info->decompressedSize += additional;
+                }
+                else{
+                    info->canComputeDecompSize = 0;
+                }
+            }
+
+            /* skip the rest of the blocks in the frame */
+            do{
+                BYTE blockHeaderBuffer[3];
+                U32 blockHeader;
+                int blockSize;
+                readBytes = fread(blockHeaderBuffer, 1, 3, srcFile);
+                if(readBytes != 3){
+                    DISPLAY("There was a problem reading the block header\n");
+                    exit(1);
+                }
+                blockHeader = MEM_readLE24(blockHeaderBuffer);
+                lastBlock = blockHeader & 1;
+                blockSize = (blockHeader - (blockHeader & 7)) >> 3;
+                fseek(srcFile, blockSize, SEEK_CUR);
+            }while(lastBlock != 1);
+            {
+                /* check if checksum is used */
+                int contentChecksumFlag = (frameHeaderDescriptor & (1 << 2)) >> 2;
+                if(contentChecksumFlag){
+                    info->usesCheck = 1;
+                }
+                if(contentChecksumFlag){
+                    fseek(srcFile, 4, SEEK_CUR);
+                }
+            }
+        }
+        else if(magicNumber==ZSTD_MAGIC_SKIPPABLE_START){
+            BYTE frameSizeBuffer[4];
+            long frameSize;
+            size_t readBytes = fread(frameSizeBuffer, 1, 4, srcFile);
+            info->numSkippableFrames++;
+            if(readBytes != 4){
+                DISPLAY("There was an error reading skippable frame size");
+                exit(1);
```

Comment:
Irrecoverable `exit()`.

In the case of a list of files to `--list`, it would be better if such a case was flagged as an `error`, but the analysis would continue onto next file.